compiler/parser.nim

```diff
@@ -1286,45 +1293,20 @@ proc primary(p: var Parser, mode: PrimaryMode): PNode =
     return
 
   case p.tok.tokType
-  of tkTuple: result = parseTuple(p, mode == pmTypeDef)
   of tkProc: result = parseProcExpr(p, mode notin {pmTypeDesc, pmTypeDef}, nkLambda)
```

Comment:
I believe this is the only part of the grammar left where `pmTypeDesc`/`pmTypeDef` (functionally only `pmTypeDesc`) are not represented. According to the grammar `proc foo(x: proc: int = 3) = discard` should parse as `proc foo(x: (proc: int = 3)) = discard`. But this is currently not worth the effort of clarifying, if someone else wants to do it that's fine with me.